CONTRIBUTOR.md

```diff
@@ -0,0 +1,231 @@
+
+### Running Tests
+
+See [tests/README.md](tests/README.md) for comprehensive testing instructions.
+
+### Test Requirements
+
+- **Unit Tests**: All new functionality must include unit tests
+- **Integration Tests**: Add integration tests for complex features
+- **Test Coverage**: Maintain or improve existing test coverage
+- **Test Naming**: Use descriptive test names that explain what is being tested
+
+### Test Structure
+
+```python
+def test_feature_should_behave_correctly_when_condition():
+    # Arrange
+    setup_data = create_test_data()
+    
+    # Act
+    result = function_under_test(setup_data)
+    
+    # Assert
+    assert result.expected_property == expected_value
+```
+
+## Submitting Changes
+
+### Pull Request Process
+
+1. **Create Feature Branch**
+   ```bash
+   git checkout -b feature/your-feature-name
+   # or
+   git checkout -b fix/issue-number-description
+   ```
+
+2. **Make Your Changes**
+   - Write clean, well-documented code
+   - Add appropriate tests
+   - Update documentation if needed
+
+3. **Test Your Changes**
+   ```bash
+   # Run tests
+   python -m pytest
+
+   # Check code style
+   flake8 src/ tests/
+
+   # Update sync code if you modified async code
+   python tools/unasync.py --check
+   ```
+
+4. **Commit Your Changes**
+   ```bash
+   git add .
+   git commit -m "Clear, descriptive commit message"
+   ```
+
+   **Commit Message Guidelines:**
+   - Use present tense ("Add feature" not "Added feature")
+   - Keep first line under 50 characters
+   - Reference issue numbers when applicable (#123)
+   - Include breaking change notes if applicable
+
+5. **Push and Create Pull Request**
+   ```bash
+   git push origin feature/your-feature-name
+   ```
+   
+   Then create a pull request through GitHub's interface.
+
+### Pull Request Guidelines
+
+- **Title**: Clear and descriptive
+- **Description**: Explain what changes you made and why
+- **Linked Issues**: Reference related issues using "Fixes #123" or "Relates to #123"
+- **Labels**: Review available issue/PR labels and apply relevant ones to help with categorization and triage
+- **Documentation**: Update documentation for user-facing changes
+- **Tests**: Include appropriate tests
+- **Breaking Changes**: Clearly mark any breaking changes
+
+### Code Review Process
+
+- All pull requests require review before merging
+- Address reviewer feedback promptly
+- Keep discussions respectful and constructive
+- Be open to suggestions and alternative approaches
+
+## Reporting Issues
+
+### Using Labels
+
+When creating issues or pull requests, please review the available labels and apply those that are relevant to your submission. This helps maintainers categorize and prioritize work effectively. Common label categories include:
+
+- **Type**: bug, enhancement, documentation, question
+- **Priority**: high, medium, low
+- **Component**: producer, consumer, admin, schema-registry
+- **Status**: needs-investigation, help-wanted, good-first-issue
```

Comment:
We have added more labels. We should update this list.

Reply:
Going to reference this is a subset of the options as I want to avoid multiple sources of truth